test/parallel/test-tls-multi-key.js

```diff
@@ -21,52 +21,168 @@
+  ],
+  cert: [
     fixtures.readKey('ec-cert.pem'),
-  ]
-};
+  ],
+  pfx: fixtures.readKey('agent1.pfx'),
+  passphrase: 'sample',
+  client: { ca: [
+    fixtures.readKey('ec-cert.pem'),
+    fixtures.readKey('ca1-cert.pem'),
+  ] },
+});
 
-const ciphers = [];
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs
+test({
+  key: [
+    fixtures.readKey('ec10-key.pem'),
+    fixtures.readKey('agent10-key.pem'),
+  ],
+  cert: [
+    fixtures.readKey('agent10-cert.pem'),
+    fixtures.readKey('ec10-cert.pem'),
+  ],
+  rsaCN: 'agent10.example.com',
+  eccCN: 'agent10.example.com',
+  client: { ca: [
+    fixtures.readKey('ca2-cert.pem'),
+    fixtures.readKey('ca5-cert.pem'),
+  ] },
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs,
+// using PFX for EC.
+test({
+  key: [
+    fixtures.readKey('agent10-key.pem'),
+  ],
+  cert: [
+    fixtures.readKey('agent10-cert.pem'),
+  ],
+  pfx: fixtures.readKey('ec10.pfx'),
+  passphrase: 'sample',
+  rsaCN: 'agent10.example.com',
+  eccCN: 'agent10.example.com',
+  client: { ca: [
+    fixtures.readKey('ca2-cert.pem'),
+    fixtures.readKey('ca5-cert.pem'),
+  ] },
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs,
+// using PFX for RSA.
+test({
+  key: [
+    fixtures.readKey('ec10-key.pem'),
+  ],
+  cert: [
+    fixtures.readKey('ec10-cert.pem'),
+  ],
+  pfx: fixtures.readKey('agent10.pfx'),
+  passphrase: 'sample',
+  rsaCN: 'agent10.example.com',
+  eccCN: 'agent10.example.com',
+  client: { ca: [
+    fixtures.readKey('ca2-cert.pem'),
+    fixtures.readKey('ca5-cert.pem'),
+  ] },
+});
+
+function test(options) {
+  const rsaCN = options.rsaCN || 'agent1';
+  const eccCN = options.eccCN || 'agent2';
+  const clientTrustRoots = options.client.ca;
+  delete options.rsaCN;
+  delete options.eccCN;
+  delete options.client;
+  const server = tls.createServer(options, function(conn) {
+    conn.end('ok');
+  }).listen(0, common.mustCall(connectWithEcdsa));
 
-const server = tls.createServer(options, function(conn) {
-  conn.end('ok');
-}).listen(0, function() {
-  const ecdsa = tls.connect(this.address().port, {
-    ciphers: 'ECDHE-ECDSA-AES256-GCM-SHA384',
-    rejectUnauthorized: false
-  }, function() {
-    ciphers.push(ecdsa.getCipher());
+  function connectWithEcdsa() {
+    const ecdsa = tls.connect(this.address().port, {
+      ciphers: 'ECDHE-ECDSA-AES256-GCM-SHA384',
+      rejectUnauthorized: true,
+      ca: clientTrustRoots,
+      checkServerIdentity: (_, c) => assert.strictEqual(c.subject.CN, eccCN),
+    }, common.mustCall(function() {
+      assert.deepStrictEqual(ecdsa.getCipher(), {
+        name: 'ECDHE-ECDSA-AES256-GCM-SHA384',
+        version: 'TLSv1/SSLv3'
+      });
+      assert.strictEqual(ecdsa.getPeerCertificate().subject.CN, eccCN);
+      // XXX(sam) certs don't currently include EC key info, so depend on
+      // absence of RSA key info to indicate key is EC.
```

Comment:
That comment gets obsoleted by #24358, right?

Reply:
Yes, it was while writing this test I realized testing `is not rsa` was the only way to test `is ec`. The comments not really obsolete until the EC SPKI lands, at which point I can rewrite the test.

Reply:
^-- @tniessen FYI.